block/gossip.go

```diff
@@ -22,6 +23,10 @@ func (m *Manager) onNewGossipedBlock(event pubsub.Message) {
 		return
 	}
 
+	if block.Header.Height > m.TargetHeight.Load() {
+		m.TargetHeight.Store(block.Header.Height)
+	}
```

Comment:
this is not concurrency safe you need to use compareAndSwap
and please use stdlib max